Add docs for tunnel

Luc45 opened the pull request:
To review:

- npm run start
- Go to "Local Test Environment -> Tunnelling"

Luc45 committed:
Add docs for tunnel

Luc45 committed:
Add documentation for custom tunnels

Luc45 committed:
Suggest Cloudflare DNS on all O.S's

Luc45 committed:
Tunnelling documentation

Luc45 committed:
Merge branch 'trunk' into 24-10/tunnel

Luc45 committed:
Tweak

zhongruige commented on docs/environment/05-tunnel.md:
````diff
@@ -0,0 +1,265 @@
+brew install cloudflared
+```
+
+### Tunnelling on Linux
````
```suggestion
### Tunneling on Linux
```

zhongruige commented on docs/environment/05-tunnel.md:
```diff
@@ -0,0 +1,265 @@
+
+No additional requirements are needed for tunneling on Linux.
+
+### Tunnelling on WSL
```
```suggestion
### Tunneling on WSL
```

zhongruige commented on docs/environment/05-tunnel.md:
```diff
@@ -0,0 +1,265 @@
+
+### Tunnelling on WSL
+
+Tunnelling on Windows Subsystem for Linux (WSL) is currently not supported. Let us know if you need this feature by [opening an issue](https://github.com/woocommerce/qit-cli/issues).
```
```suggestion
Tunneling on Windows Subsystem for Linux (WSL) is currently not supported. Let us know if you need this feature by [opening an issue](https://github.com/woocommerce/qit-cli/issues).
```

zhongruige commented on docs/environment/05-tunnel.md:
```diff
@@ -0,0 +1,265 @@
+
+## Using Tunneling with QIT CLI Commands
+
+You can use tunnelling in the commands that utilize the QIT CLI environment capabilities. Currently, the following commands support tunneling:
```
```suggestion
You can use tunneling in the commands that utilize the QIT CLI environment capabilities. Currently, the following commands support tunneling:
```

zhongruige commented on docs/environment/05-tunnel.md:
```diff
@@ -0,0 +1,265 @@
+- **cloudflared-docker**: Cloudflare Tunnel via Docker container.
+- **cloudflared-binary**: Cloudflare Tunnel using the local binary.
+- **cloudflared-persistent**: Pre-configured persistent Cloudflare Tunnel.
+- **jurassictube**: JurassicTube Tunnel (Automattic employees only).
```
I'm thinking we can drop this from our public docs and just add in a documentation page, similar to the authentication methods in Pe98dy-Ir-p2

zhongruige commented on docs/environment/05-tunnel.md:
```diff
@@ -0,0 +1,265 @@
+
+# Comparison Table
+
+| Feature                          | cloudflared-docker | cloudflared-binary | cloudflared-persistent | jurassictube | custom  |
```
Since everything is `Depends` for custom, do you think we should keep this column?

Also suggestion to drop the JurassicTube and Automattic Only from this table, too.

zhongruige commented on docs/environment/05-tunnel.md:
```diff
@@ -0,0 +1,265 @@
+[cloudflared-docker] Cloudflare Tunnel via Docker container
+[cloudflared-binary] Cloudflare Tunnel using the local binary
+[cloudflared-persistent] Persistent Cloudflared Tunnel
+[jurassictube] JurassicTube Tunnel (Automattic employees only)
```
I'm unsure about showing this unless the user has the ability to access it.

zhongruige commented on docs/environment/05-tunnel.md:
```diff
@@ -0,0 +1,265 @@
+
+## Implementing a Custom Tunnel
+
+If you need to use a custom tunnelling method, refer to the [Custom Tunneling Methods](./custom-tunnel) section for instructions on how to create and use a custom tunneling method.
```
```suggestion
If you need to use a custom tunneling method, refer to the [Custom Tunneling Methods](./custom-tunnel) section for instructions on how to create and use a custom tunneling method.
```

zhongruige reviewed (commented):
Left a few comments on some typos and a suggestion around Jurassic Tube, other than those everything else is looking fantastic on these docs, thank you for being so thorough and clear in this documentation!

Luc45 committed:
Update docs/environment/05-tunnel.md

Co-authored-by: Greg <71906536+zhongruige@users.noreply.github.com>

Luc45 committed:
Update docs/environment/05-tunnel.md

Co-authored-by: Greg <71906536+zhongruige@users.noreply.github.com>

Luc45 committed:
Update docs/environment/05-tunnel.md

Co-authored-by: Greg <71906536+zhongruige@users.noreply.github.com>

Luc45 committed:
Update docs/environment/05-tunnel.md

Co-authored-by: Greg <71906536+zhongruige@users.noreply.github.com>

Luc45 committed:
Update docs/environment/05-tunnel.md

Co-authored-by: Greg <71906536+zhongruige@users.noreply.github.com>

Luc45 commented:
>  Since everything is Depends for custom, do you think we should keep this column?
> Also suggestion to drop the JurassicTube and Automattic Only from this table, too.

The reason for keeping `custom` tunneling and Jurassic Tube is a similar one: Clarity that they exist.

I think merely documenting support for Jurassic Tube is not a concern. We are not providing a download link, it requires proxy and authentication to be used anyway. We are just mentioning support for it. Given that the majority of our users are Automattic anyway, it seems reasonable. But I'm open to double check with someone else, what do you think?

Luc45 committed:
Typo

zhongruige commented:
Maybe we could remove references to `Automattic employees only` and just leave it as an option without going into detail on access to it, that way it's an even middle ground.

Luc45 commented:
> Maybe we could remove references to Automattic employees only

Isn't it more confusing to the user?

zhongruige commented:
I can see that, so we could keep it, mostly I would also say it feels a bit weird to share a tool they can't use in general in the public docs like this (and since it's internal if we should even be sharing it publicly at all)

Luc45 committed:
Remove Jurassic Tube mentions from the documentation, remove "custom"

Luc45 commented:
@zhongruige I've removed mentions of Jurassic Tube from the documentation and created this internal documentation page instead: https://masamunep2.wordpress.com/documentation/qit/jurassic-tube-tunnel-documentation/

I've also removed the "Custom" column and added a "PS" instead.

Let me know what you think!

zhongruige reviewed (approved):
Changes look great! This is awesome to have in, and appreciate the documentation updates @Luc45! :shipit:

Luc45 committed:
Last but not least, move "Using a Persistent Tunnel" to it's own page
